build/build.c

```diff
@@ -172,9 +179,67 @@ rpmRC doScript(rpmSpec spec, rpmBuildFlags what, const char *name,
     return rc;
 }
 
-static rpmRC buildSpec(BTA_t buildArgs, rpmSpec spec, int what)
+static rpmRC doBuildRequires(rpmSpec spec, int test)
+{
+    StringBuf sb_stdout = NULL;
+    int outc;
+    ARGV_t output = NULL;
+
+    rpmRC rc = 1; /* assume failure */
+
+    if (!spec->buildrequires) {
+	rc = RPMRC_OK;
+	goto exit;
+    }
+
+    if ((rc = doScript(spec, RPMBUILD_BUILDREQUIRES, "%generate_buildrequires",
+		       getStringBuf(spec->buildrequires), &sb_stdout, test)))
+	goto exit;
+
+    /* add results to requires of the srpm */
+    argvSplit(&output, getStringBuf(sb_stdout), "\n\r");
+    outc = argvCount(output);
+
+    if (!outc) {
+	rc = RPMRC_OK;
+	goto exit;
+    }
+
+    for (int i = 0; i < outc; i++) {
+	parseRCPOT(spec, spec->sourcePackage, output[i], RPMTAG_REQUIRENAME, 0, 0, addReqProvPkg, NULL);
+    }
+
+    rpmdsPutToHeader(*packageDependencies(spec->sourcePackage, RPMTAG_REQUIRENAME), spec->sourcePackage->header);
+
+    parseRCPOT(spec, spec->sourcePackage, "rpmlib(DynamicBuildRequires) = 4.15.0-1", RPMTAG_PROVIDENAME, 0, RPMSENSE_FIND_PROVIDES | RPMSENSE_RPMLIB, addReqProvPkg, NULL);
```

Comment:
There are some seriously overlong lines here, both parseRCPOT() calls and the rpmdsPutToHeader(). Please adjust to 80 character line-length.